### PR TITLE
Alec

### PR DESCRIPTION
**Pull request by AMLeng**

### CODE REVIEW

**Comment by edelmanjm** [comment]
Should remove the device lock.

File: steer.py [hunk 1]
```diff
@@ -50,255 +52,230 @@
...
-  return var
+    """Helper to create a Variable stored on CPU memory.
+    Args:
+        name: name of the variable
+        shape: list of ints
+        initializer: initializer for Variable
+    Returns:
+        Variable Tensor
+    """
+    with tf.device('/cpu:0'):
```
Comment (edelmanjm) [inline steer.py]: Why need device lock?

File: steer.py [hunk 2]
```diff
@@ -50,255 +52,230 @@
...
+            name,
+            shape,
+            tf.truncated_normal_initializer(stddev=stddev, dtype=dtype))
+    if wd is not None:
+        weight_decay = tf.multiply(tf.nn.l2_loss(var), wd, name='weight_loss')
+        tf.add_to_collection('losses', weight_decay)
+    return var
 
 
 def distorted_inputs():
```
Comment (edelmanjm) [inline steer.py]: Do we want to location to be coded this way instead of passed into the method?

File: steer.py [hunk 3]
```diff
@@ -50,255 +52,230 @@
...
+        images: Images. 4D tensor of [batch_size, IMAGE_SIZE, IMAGE_SIZE, 3] size.
+        labels: Labels. 1D tensor of [batch_size] size.
+    Raises:
+        ValueError: If no data_dir
+    """
+    images, labels = steer_input.distorted_inputs(data_dir=DATA_DIR,batch_size=BATCH_SIZE)
+    return images, labels
 
 
 def inputs():
```
Comment (edelmanjm) [inline steer.py]: Again, do we want to location to be coded this way instead of passed into the method?

File: steer.py (hunk 1 above)
Comment (edelmanjm) [inline steer.py]: Maybe add more comments explaining why?

File: steer_eval.py [hunk 4]
```diff
@@ -35,113 +35,94 @@
 import time
 import sys
 
+import os
+
+os.environ['TF_CPP_MIN_LOG_LEVEL'] = '2'
```
Comment (edelmanjm) [inline steer_eval.py]: What dis do?

File: steer_eval.py [hunk 5]
```diff
@@ -35,113 +35,94 @@
...
+                saver.restore(sess,ckpt.model_checkpoint_path)
+                global_step = ckpt.model_checkpoint_path.split('/')[-1].split('-')[-1]
+                print(global_step)
+                sys.stdout.flush()
+            else:
+                print('No checkpoint file found')
+                return
+
+            """
+            tf.train.start_queue_runners(sess)
```
Comment (edelmanjm) [inline steer_eval.py]: Remove unnecessary commented out code

File: steer_eval.py [hunk 6]
```diff
@@ -35,113 +35,94 @@
...
+                total_sample_count = num_iter * BATCH_SIZE
+                step = 0
+                while step < num_iter and not coord.should_stop():
+                     predictions = sess.run([loss_op])
+                     error_sum += np.sum(predictions)
+                     step += 1
+                precision = error_sum / total_sample_count
+                print('%s: Average Error = %.3f' % (datetime.now(), precision))
+
+            except Exception as e:    # pylint: disable=broad-except
```
Comment (edelmanjm) [inline steer_eval.py]: This seems dangerous

File: steer_eval.py [hunk 7]
```diff
@@ -35,113 +35,94 @@
...
+            # Start the queue runners.
+            coord = tf.train.Coordinator()
+            try:
+                threads = []
+                for qr in tf.get_collection(tf.GraphKeys.QUEUE_RUNNERS):
+                    threads.extend(qr.create_threads(sess, coord=coord, daemon=True,start=True))
+
+                """
+                # Calculate loss.
+                loss=sess.run(loss_op)
```
Comment (edelmanjm) [inline steer_eval.py]: More commented out code

File: steer_eval.py [hunk 8]
```diff
@@ -35,113 +35,94 @@
...
+            tf.train.start_queue_runners(sess)
+            print(str(datetime.now()) + " " + sess.run(loss_op))
+            sys.stdout.flush()
+            """
+            # Start the queue runners.
+            coord = tf.train.Coordinator()
+            try:
+                threads = []
+                for qr in tf.get_collection(tf.GraphKeys.QUEUE_RUNNERS):
+                    threads.extend(qr.create_threads(sess, coord=coord, daemon=True,start=True))
```
Comment (edelmanjm) [inline steer_eval.py]: Don't entirely understand what's going on here with the threads. Comments maybe?

File: steer_input.py [hunk 9]
```diff
@@ -29,160 +29,154 @@
 #TODO:
 #Allow for flipping image distortion (that is, modify the label accordingly). Similarly, fancier augmentations of the data also require altering the labels
 
-NUM_EXAMPLES_PER_EPOCH_FOR_TRAIN = 15000
-NUM_EXAMPLES_PER_EPOCH_FOR_EVAL = 10000
+NUM_EXAMPLES_PER_EPOCH_FOR_TRAIN = 150
```
Comment (edelmanjm) [inline steer_input.py]: Seems awfully low?

File: steer_input.py [hunk 10]
```diff
@@ -29,160 +29,154 @@
...
-  else:
-    images, label_batch = tf.train.batch(
-        [image, label],
-        batch_size=batch_size,
-        num_threads=16,
-        capacity=min_queue_examples + 3 * batch_size)
-
-  return images, tf.reshape(label_batch, [batch_size])
+    class ImageRecord(object):
+        pass
```
Comment (edelmanjm) [inline steer_input.py]: Why?

Comment (edelmanjm) [inline steer_input.py]: Seems unusual to have a class that does nothing

File: steer.py (hunk 1 above)
Comment (AMLeng) [inline steer.py]: They're covered in the CNN tutorial, but I guess I should

File: steer_input.py (hunk 9 above)
Comment (AMLeng) [inline steer_input.py]: These numbers don't really matter. I set them low so that it'd be quicker

**Pull request merged**